packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -428,6 +428,137 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   String toString() => '${objectRuntimeType(this, 'TransitionRoute')}(animation: $_controller)';
 }
 
+/// A mixin used by routes to allow the top route define how the bottom route
+/// will animate when the top route enters and exits.
+///
+/// When a [Navigator] is instructed to pop/push, the bottom route will check if the
+/// top route should define the secondary animation transition for this bottom route
+/// and in that case it will delegate that transition
+mixin DelegatedTransitionsRoute<T> on TransitionRoute<T> {
```

Comment:
Why `on TransitionRoute`? It seems we require `buildTransitions` to be implemented, which is coming from ModalRoute?

Reply:
Because `ModalRoute` is the one that implements `DelegatedTransitionsRoute`. Otherwise all this could be implemented inside `ModalRoute`